doc/pipeline-diagram.md

```diff
@@ -0,0 +1,250 @@
+    end note
+    specs: Specs
+    note left of specs
+        Generate an OBuilder spec which is an equivalence of
+        taking a subset of the set generate in the previous
+        step.
+    end note
+    extract: Extract results
+    note left of extract
+        For each package, it extracts the result computed
+        from the jobs.
+    end note
+    union: Union
+    note left of union
+        Merge all the packages to get a mega universe.
+    end note
+    bless: Bless packages
+    note left of bless
+        For each package, it selects one universe:
+        the one with the maximal number of dependencies.
+    end note
+    odoc: ODoc generation
+    note left of odoc
+        For each Prep.t, it schedules a job to run `odoc` to
+        build the documentation on a remote worker.
+    end note
+    html: Update Html
+    note left of html
+        Update, in an atomic way, the link to the
+        documentation to avoid breaking.
+    end note
+
+    
+
+    state Server {
+        [*] --> opam
+        opam --> track: git hash
+        track --> solver: { package, digest } list
+        solver --> set: { opam package, universe, commit} list
+        set --> jobs: { opam package, universe, commit } set
+        jobs --> specs: { package, package list} list
+        specs --> extract: { base, hash, package, result } map
+        extract --> union: { base, hash, package, result} map
+        union --> bless: { base, hash, package, result} map
+        bless --> odoc: { package, universe } map
+        odoc --> html: { package, html } list
+        html --> [*]: { package, blessing, hash } map
+    }
+
+    state Jobs {
+        job_1: descr 1
+        note right of job_1
+            Install Voodo
+            Install package from within the universe
+            Execute Voodoo
+            Rsync the result and compute the hash
+        end note
+        job_2: desrc 2
+        note right of job_2
+            Download the computed tar
+            Build the doc
+        end note    
+    }
+
+    job_1 --> specs: data hash or FAILED
+    specs --> job_1: job specification
+    job_2 --> odoc: job specification
+    odoc --> job_2: odoc directories hashes
+```
+
+### Track the packages from opam
+
+```mermaid
+stateDiagram-v2
+    opam: Opam_repo
+    note left of opam
+        Get the commit hash from the Github repository
+        and catch commit for Voodoo.
+    end note
+    track: Track
+    note left of track
+        Compute the list of the available package in
+        opam with their version.
+    end note
+    
+    
+    [*] --> opam
+    opam --> track: git hash
+    track --> [*]: { package, digest } list
+```
+
+### Compute the set of dependencies
+
+```mermaid 
+stateDiagram-v2
+    solver: Solver
+    note left of solver
+        For each tuple (package, version), it computes
+        the dependencies.
+        Two solver steps:
+        - with ocaml-base-compiler >= 4.02.3
```

Comment:
Currently there's also an upper bound on ocaml 5, as the preparation step currently doesn't work